iib/web/models.py

```diff
@@ -190,6 +204,79 @@ def add_state(self, state, state_reason):
         db.session.flush()
         self.request_state_id = request_state.id
 
+    def to_json(self, verbose=True):
+        states = [
+            {
+                'state': RequestStateMapping(state.state).name,
+                'state_reason': state.state_reason,
+                'updated': state.updated.isoformat(),
+            }
+            for state in self.states
+        ]
+        # Reverse the list since the latest states should be first
+        states = list(reversed(states))
+        latest_state = states[0]
+
+        rv = {
```

Comment:
Could you alphabetize the keys to make them easier to read?

Reply:
This is missing the following keys:
* from_index_resolved
* binary_image_resolved
* index_image
* request_type

Please note that these will be `None` when the request is first created, so you will need to guard against this.

Reply:
In the API document, we said we'd have an `arches` list, which would map to the `index_image` arches.

Reply:
Regarding the `arches_list`. Shouldn't it be part of the `index_image`? Or should we have the arches in the `index_image` json and have the `arches_list`?

Reply:
Ok, I see you prefer to have only the pull_spec for all the images. I'll change it. :)